compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp

```diff
@@ -0,0 +1,188 @@
+    qparam->scale = quantparam->scale;
+    qparam->zerop = quantparam->zerop;
+    qparam->min = quantparam->min;
+    qparam->max = quantparam->max;
+    qparam->quantized_dimension = quantparam->quantized_dimension;
+
+    cloned->quantparam(std::move(qparam));
+  }
+  // sparsity
+  const auto *sparsity = node->sparsityparam();
+  if (sparsity != nullptr)
+  {
+    auto sparam = std::make_unique<luci::SparsityParam>();
+    sparam->traversal_order = sparsity->traversal_order;
+    sparam->block_map = sparsity->block_map;
+    sparam->dim_metadata = sparsity->dim_metadata;
+
+    cloned->sparsityparam(std::move(sparam));
+  }
+  // op version
+  cloned->op_version(node->op_version());
+
+  return cloned;
+}
+
+bool forward_reshape(luci::CircleReshape *reshape, luci::CircleNeg *neg)
+{
+  assert(reshape != nullptr);
+  assert(neg != nullptr);
+
+  luci::CircleConst *cloned_shape = nullptr;
+  const auto reshape_shape = dynamic_cast<luci::CircleConst *>(reshape->shape());
+  // only support CircleConst for now
+  if (reshape_shape == nullptr)
+    return false;
+
+  if (reshape_shape->dtype() == loco::DataType::S32)
+    cloned_shape = clone<loco::DataType::S32>(reshape_shape);
+  else if (reshape_shape->dtype() == loco::DataType::S64)
+    cloned_shape = clone<loco::DataType::S64>(reshape_shape);
+  else
+    return false;
+
+  loco::Graph *graph = neg->graph();
+  // create reshape placed after neg
+  luci::CircleReshape *new_reshape = graph->nodes()->create<luci::CircleReshape>();
+  auto ns_rank = reshape->newShape()->rank();
+  new_reshape->newShape()->rank(ns_rank);
+  for (uint32_t r = 0; r < ns_rank; ++r)
+    new_reshape->newShape()->dim(r) = reshape->newShape()->dim(r);
+
+  new_reshape->shape(cloned_shape);
+
+  // reconnect network
+  loco::replace(neg).with(new_reshape);
+  neg->x(reshape->tensor());
+  new_reshape->tensor(neg);
+
+  // need to reset shape as it leaped over Reshape
+  // TODO Remove loco::shape_erase()
+  loco::shape_erase(neg);
+  neg->shape_status(luci::ShapeStatus::UNDEFINED);
+
+  return true;
+}
+
+class ForwardReshape final : public luci::CircleNodeMutableVisitor<bool>
+{
+protected:
+  bool visit(luci::CircleNode *node)
+  {
+    LOGGER(l);
+    INFO(l) << "ForwardReshape: Unsupported operator: " << node->name() << std::endl;
+    return false;
+  }
+
+  bool visit(luci::CircleNeg *node)
+  {
+    auto reshape = as_reshape(node->x());
+    if (reshape == nullptr)
+      return false;
+    return forward_reshape(reshape, node);
+  }
+
+  // TODO add more unary operators
+};
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * BEFORE
+ *                       |
+ *                  [CircleNode]  [CircleConst]
+ *                       |       /
+ *                 [CircleReshape]
+ *                /      |
+ *     [CircleNode]  [(UnaryOp)]
+ *          |            |     \
+ *          |            |      [CircleNode]
+ *          |            |           |
```

Comment:
what are these `CircleNode`?
```suggestion
 *                 [CircleReshape]
 *                       |
 *                    [(UnaryOp)]
 *                       |     
 *                       |      
 *                       |           
```

Reply:
They can be any CircleNode that is connected to our target node that I wanted to considered while graph connection change.